Bandhan majumder/issue 3522 profile

bandhan-majumder opened the pull request:
<!--
This section can be deleted after reading.

We employ the following branching strategy to simplify the development process and to ensure that only stable code is pushed to the `master` branch:

- `develop`: For unstable code: New features and bug fixes.
- `master`: Where the stable production ready code lies. Only security related bugs.

NOTE!!!

ONLY SUBMIT PRS AGAINST OUR `DEVELOP` BRANCH. THE DEFAULT IS `MAIN`, SO YOU WILL HAVE TO MODIFY THIS BEFORE SUBMITTING YOUR PR FOR REVIEW. PRS MADE AGAINST `MAIN` WILL BE CLOSED.

-->

<!--
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request.
-->

**What kind of change does this PR introduce?**

<!-- E.g. a bugfix, feature, refactoring, etc… -->

**Issue Number:**

Fixes #<!--Add related issue number here.-->

**Snapshots/Videos:**

<!--Add snapshots or videos wherever possible.-->

**If relevant, did you update the documentation?**

<!--Add link to Talawa-Docs.-->

**Summary**

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->

**Does this PR introduce a breaking change?**

<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->

## Checklist

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [ ] I have written tests for all new changes/features
- [ ] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass


**Other information**

<!--Add extra information about this PR here-->

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

<!--Yes or No-->

bandhan-majumder commented:
writing tests and fixing failing steps.

palisadoes commented:
Please change the title so that it's more searchable

palisadoes commented:
1. Please use your initiative to find a solution that maintains functionality.
2. Can you get the information using multiple queries?
3. Can you create an updated query to get the information?

Create a suitable solution. Discuss it in the slack channels

palisadoes commented:
1. Disable the event and tags features for this PR
2. Create issues to restore the functionality separately
3. When complete let us know when this is ready for review

bandhan-majumder commented on src/utils/languages.ts:
```diff
@@ -17,7 +17,7 @@ const languages = [
   },
   {
-    code: 'sp',
+    code: 'es',
```
It was needed to match the enum as graphql was not accepting sp as languageCode

bandhan-majumder commented:
> 1. Disable the event and tags features for this PR
> 
>     2. Create issues to restore the functionality separately
> 
>     3. When complete let us know when this is ready for review

@palisadoes . Sure. I have refactored the code which are currently already implemented and removed which are not but was still there in the same component. (tags, events..). Currently working on the coderabbit's suggestions

bandhan-majumder commented on schema.graphql:
```diff
@@ -909,8 +910,7 @@ type Mutation {
   updatePost(data: PostUpdateInput, id: ID!): Post!
   updateSessionTimeout(timeout: Int!): Boolean!
-  updateUserPassword(data: UpdateUserPasswordInput!): UserData!
-  updateUserProfile(data: UpdateUserInput, file: String): User!
+  updateUser(input: MutationUpdateUserInput!): UserPg
```
done

bandhan-majumder commented on src/components/UserPortal/UserProfile/EventsAttendedByUser.tsx:
```diff
@@ -10,21 +10,26 @@ import EventsAttendedByMember from 'components/MemberDetail/EventsAttendedByMemb
+    addressLine2: string;
+    emailAddress: string;
+    birthDate: null | string;
+    city: string;
+    avatar: File | null;
+    countryCode: string;
+    description: string;
+    educationGrade: string;
+    employmentStatus: string;
+    homePhoneNumber: string;
     maritalStatus: string;
-    address: string;
+    mobilePhoneNumber: string;
+    name: string;
+    natalSex: string;
+    naturalLanguageCode: string;
+    password: string;
+    postalCode: string;
     state: string;
-    country: string;
-    image: string;
-    eventsAttended: { _id: string }[];
+    workPhoneNumber: string;
```
this is for future reference. removing it may cause confusion

bandhan-majumder commented on src/screens/MemberDetail/MemberDetail.tsx:
```diff
@@ -471,7 +391,43 @@
+                    }
+                  />
+                </Col>
+                <Col md={12}>
+                  <label htmlFor="password" className="form-label">
+                    {tCommon('password')}
+                  </label>
+                  <input
+                    id="password"
+                    value={formState.password}
+                    className={`form-control ${styles.inputColor}`}
+                    type="password"
+                    name="password"
+                    onChange={(e) =>
+                      handleFieldChange('password', e.target.value)
+                    }
+                    placeholder="* * * * * * * *"
+                  />
+                </Col>
```
it is done on the backend

bandhan-majumder commented on src/screens/MemberDetail/MemberDetailMocks.ts:
```diff
@@ -1,534 +1,189 @@
   {
     request: {
-      query: USER_DETAILS,
+      query: CURRENT_USER,
```
that will be solved in other prs. It is out of scope for this issue

palisadoes commented:
Please fix the failing tests

bandhan-majumder commented:
Sure @palisadoes . I was testing the file upload feature. I am able to store it in minio and postgres after a few iterations. Just have to implement it in frontend now.

for reference: 
look at the 15th entry
![image](https://github.com/user-attachments/assets/4af956c9-9c8c-4e45-8e41-317097762ac7)

bandhan-majumder commented:
@palisadoes 
1. while sending files, I am using multipart uploads with graphql mutations and fetch which I have send a form-data.
2. that fetch call needs authentication where I am sending my token directly from localstorage
3. while using that, my this check failing
![image](https://github.com/user-attachments/assets/21f6312d-80ca-46c7-bd30-e7a23842546b)

should I just ignore it? Using only graphql, I am unable to send multipart request. So I need to use regular `fetch`

bandhan-majumder commented:
@palisadoes do i need to commit the auto changed .md files too?

bandhan-majumder commented:
@palisadoes I am closing this pr and opening a new fresh one. In this pr, .md files are also not included which may cause confusions on documentation.
